include/engine/unpacking_cache.hpp

```diff
@@ -0,0 +1,61 @@
+#ifndef UNPACKING_CACHE_HPP
+#define UNPACKING_CACHE_HPP
+
+#include <boost/optional/optional_io.hpp>
+
+#include "../../third_party/compute_detail/lru_cache.hpp"
+#include "util/typedefs.hpp"
+
+namespace osrm
+{
+namespace engine
+{
+class UnpackingCache
+{
+  private:
+    boost::compute::detail::lru_cache<std::tuple<NodeID, NodeID, std::size_t>, EdgeDuration> cache;
```

Comment:
Instead of using `std::size_t` (8 bytes) you can use `unsigned char` (1 byte) for `exclude_index` here since its maximum values is 8.